src/app/notes/course_teacher/course_teacher.component.ts

```diff
@@ -73,6 +76,65 @@ export class CourseTeacherInputWindow implements OnInit {
             this.getCourses();
         });
     }
+
+    exportGradesAsCSV() {
+        this.notesService.exportGradesAsCSV().subscribe(res => {
+            const gradesAsCSV = res as GradesAsCSV[];
+            if (!gradesAsCSV)
+                return alert("Erreur. La récupération des notes à échouée.")
+            if (gradesAsCSV.length == 0)
+                return alert("Erreur. La liste des cours est vide.")
+            this.createCSV(gradesAsCSV);
+        });
```

Comment:
no error handling if the request fails
```suggestion
        this.notesService.exportGradesAsCSV().subscribe(res => {
            const gradesAsCSV = res as GradesAsCSV[];
            if (!gradesAsCSV)
                return alert("Erreur. La récupération des notes à échouée.")
            if (gradesAsCSV.length == 0)
                return alert("Erreur. La liste des cours est vide.")
            this.createCSV(gradesAsCSV);
        }, err => {
            // error handling here
        });
```